AdvancedDLSupport.AOT.Tests/Tests/Integration/NativeLibraryBuilderTests.cs

```diff
@@ -31,25 +34,53 @@ public class NativeLibraryBuilderTests
     {
         public class DiscoverCompiledTypes : NativeLibraryBuilderTestBase
         {
+            private static readonly Action ClearCache;
+
+            static DiscoverCompiledTypes()
+            {
+                var clearMethod = typeof(IDictionary).GetMethod(
+                    nameof(IDictionary.Clear),
+                    BindingFlags.Public | BindingFlags.Instance);
+                var typeCacheField = typeof(NativeLibraryBuilder).GetField(
+                    "TypeCache", BindingFlags.NonPublic | BindingFlags.Static);
+
+                if (typeCacheField == null || clearMethod == null)
+                {
+                    Assert.True(false, "ClearCache discovery needs to be fixed!");
+                    return;
+                }
+
+                ClearCache = Expression.Lambda<Action>(Expression.Call(Expression.Field(null, typeCacheField), clearMethod)).Compile();
+            }
+
             [Fact]
             public void CanDiscoverPrecompiledTypes()
             {
+                ClearCache();
```

Comment:
Move this call from all the tests to the constructor of the test class. The constructor runs once before each test.